### DISCUSSION TITLE
Using two Harmonic synths in a dag, in a .gin

### DISCUSSION THREAD

**Discussion by juanalonso** [magenta/ddsp#331] [Q&A] [answered]
Hello!

I'm experimenting a bit with the dag architecture. I'm using a gin file (I need that) and I'm trying to use two Harmonic synths. I cannot find a way to name them with unique names. I know I can do this `Harmonic.name = 'harmonic'` , but since I have two synths, I do not know how to reference each one. Of course, I can do that stringing the processors in Python, but I would love to have that definition in a gin file. Is there a way to do that? I am probably missing something trivial.

```
ProcessorGroup.dag = [

  #Can I give a name to this processor?
  (@synths.Harmonic(),  
    ['amps_1', 'harmonic_distribution_1', 'f0_hz_1']

  #And also to this one
  (@synths.Harmonic(), 
    ['amps_2', 'harmonic_distribution_2', 'f0_hz_2']

  #So I can do something like this
  (@processors.Add(), ['harmonic_1/signal', 'harmonic_2/signal']),
]
```
Thank you very much

**Comment by jesseengel** [comment, marked as answer]
Hi, you can specify multiple instances by scoping within Gin, kind of like a web address. So for your example it would look like:

```
ProcessorGroup.dag = [

  #Can I give a name to this processor?
  (@harm1/Harmonic(),  
    ['amps_1', 'harmonic_distribution_1', 'f0_hz_1']

  #And also to this one
  (@harm2/Harmonic(), 
    ['amps_2', 'harmonic_distribution_2', 'f0_hz_2']

  #So I can do something like this
  (@processors.Add(), ['harmonic_1/signal', 'harmonic_2/signal']),
]

harm1/Harmonic.name = 'harmonic_1'
harm2/Harmonic.name = 'harmonic_2'
```

With scoping you can also change the other parameters of each instance, to have separate instances, with one instance per a scoping namespace.
